p2panda-rs/src/test_utils/mocks/utils.rs

```diff
@@ -3,4 +3,4 @@
 //! Utilities for the `mocks` module.
 
 /// A custom `Result` type to be able to dynamically propagate `Error` types.
-pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
```

Comment:
Isn't `std::error::Error` always `Sync` and `Send`? https://doc.rust-lang.org/src/std/io/error.rs.html#496

Reply:
hmm...

I get this error without it:

```
error[E0277]: the size for values of type `dyn StdError + Sync + std::marker::Send` cannot be known at compilation time
   --> src/test_utils/mocks/node.rs:262:15
    |
262 |         .await?
    |               ^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `dyn StdError + Sync + std::marker::Send`
    = help: the following other types implement trait `FromResidual<R>`:
              <std::result::Result<T, F> as FromResidual<Yeet<E>>>
              <std::result::Result<T, F> as FromResidual<std::result::Result<Infallible, E>>>
    = note: required because of the requirements on the impl of `StdError` for `Box<dyn StdError + Sync + std::marker::Send>`
    = note: required because of the requirements on the impl of `From<Box<dyn StdError + Sync + std::marker::Send>>` for `Box<dyn StdError>`
    = note: required because of the requirements on the impl of `FromResidual<std::result::Result<Infallible, Box<dyn StdError + Sync + std::marker::Send>>>` for `std::result::Result<(), Box<dyn StdError>>`
```

Reply:
Maybe this is resolved now? Unless you are thinking of a larger review of the error handling patterns? I think it's fairly sensible now that this issue with the `Result` type is sorted.

Reply:
Very happy to raise a separate issue if there are still things you'd like to be looked at.

Reply:
> Maybe this is resolved now? Unless you are thinking of a larger review of the error handling patterns? I think it's fairly sensible now that this issue with the `Result` type is sorted.

Okay I think I understand now, we don't use `anyhow::Error` in `p2panda_rs` (which requires the inner `E` type to be `Sync`, `Send` and `'static`), but use `thiserror` to implement the `std::error::Error` trait which doesn't. We need them to be thread-safe though, so we introduce our own generic error type. So yeah, I think this is fine as is .. a nice refactoring maybe (in a separate issue) would be to define a lib-level export of our `Error` type which is then used across the crate, but thats a minor change.